src/include/checkpoint.h

```diff
@@ -10,26 +10,29 @@
 
 #define WT_CHECKPOINT_SESSION_FLAGS (WT_SESSION_CAN_WAIT | WT_SESSION_IGNORE_CACHE_SIZE)
 
-#define WT_CHECKPOINT_STATE_INACTIVE 0
-#define WT_CHECKPOINT_STATE_APPLY_META 1
-#define WT_CHECKPOINT_STATE_APPLY_BTREE 2
-#define WT_CHECKPOINT_STATE_UPDATE_OLDEST 3
-#define WT_CHECKPOINT_STATE_SYNC_FILE 4
-#define WT_CHECKPOINT_STATE_EVICT_FILE 5
-#define WT_CHECKPOINT_STATE_BM_SYNC 6
-#define WT_CHECKPOINT_STATE_RESOLVE 7
-#define WT_CHECKPOINT_STATE_POSTPROCESS 8
-#define WT_CHECKPOINT_STATE_HS 9
-#define WT_CHECKPOINT_STATE_HS_SYNC 10
-#define WT_CHECKPOINT_STATE_COMMIT 11
-#define WT_CHECKPOINT_STATE_META_CKPT 12
-#define WT_CHECKPOINT_STATE_META_SYNC 13
-#define WT_CHECKPOINT_STATE_ROLLBACK 14
-#define WT_CHECKPOINT_STATE_LOG 15
-#define WT_CHECKPOINT_STATE_RUNNING 16
-#define WT_CHECKPOINT_STATE_ESTABLISH 17
-#define WT_CHECKPOINT_STATE_START_TXN 18
-#define WT_CHECKPOINT_STATE_CKPT_TREE 19
+/* Checkpoint state ordered by appearance in txn_ckpt.c. Inactive should always be 0. */
+enum Checkpoint_State {
```

Comment:
I see in the WiredTiger code that we usually define enums following this syntax:
```
typedef enum {
...
} WT_CHECKPOINT_STATE;

Reply:
It's a mix. As far as I can tell, the pattern is that we use a `typedef` when we need to refer to it as a type. For a macro-ish use case, there's plenty of plain enums (c.f. `WT_CKPT_NONE` and `WT_CHILD_IGNORE`. So I think this declaration doesn't need a name at all :-)

Reply:
re-defined in: [e5e713e](https://github.com/wiredtiger/wiredtiger/pull/10733/commits/e5e713e5402219959644f91e669bc0e2c9276cd8)

Reply:
Ah, anonymous enums are C++-only. Best to stick with your suggestion @etienneptl.